test/k8sT/Services.go

```diff
@@ -412,6 +412,144 @@ var _ = Describe("K8sServicesTest", func() {
+			}
+
+			By("Checking traffic goes to local backend")
+			testCases := []struct {
+				selector string
+				pod      string
+				want     string
+				notWant  string
+			}{
+				{
+					selector: "id=app1",
+					pod:      fe1Name,
+					// Expects to see local backend name in returned Hostname field
+					want: be1Name,
+					// Expects never to see remote backend name in returned Hostname field
+					notWant: be2Name,
+				},
+				{
+					selector: "id=app2",
+					pod:      fe2Name,
+					want:     be2Name,
+					notWant:  be1Name,
+				},
+			}
+			for _, tc := range testCases {
+				for try := 0; try < repeat; try++ {
+					res, err := kubectl.ExecInPods(ctx, helpers.DefaultNamespace, tc.selector, curlTCP)
+					Expect(err).To(BeNil(), "http curl failed in %s pod", tc.selector)
+					res[tc.pod].ExpectContains(tc.want, "tcp curl does not contain expected: %s", res[tc.pod].Stdout())
+					res[tc.pod].ExpectDoesNotContain(tc.notWant, "tcp curl contains unwanted: %s", res[tc.pod].Stdout())
+					res, err = kubectl.ExecInPods(ctx, helpers.DefaultNamespace, tc.selector, curlUDP)
+					Expect(err).To(BeNil(), "tftp curl failed in %s pod", tc.selector)
+					res[tc.pod].ExpectContains(tc.want, "udp curl does not contain expected: %s", res[tc.pod].Stdout())
+					res[tc.pod].ExpectDoesNotContain(tc.notWant, "udp curl contains unwanted: %s", res[tc.pod].Stdout())
+				}
+			}
+		})
+
+		It("LRP restores service when removed", func() {
+			_ = kubectl.Delete(lrpSvcYAML)
+			// Basic sanity check
+			ciliumPods, err := kubectl.GetCiliumPods()
+			Expect(err).To(BeNil(), "Cannot get cilium pods")
+			for _, pod := range ciliumPods {
+				service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, fmt.Sprintf("cilium service list | grep %s", svcIP), "Cannot retrieve services on cilium pod")
+				service.ExpectContains("ClusterIP", "Original service is not present in the cilium service list")
+			}
+
+			By("Checking traffic goes to both backends")
+			testCases := []struct {
+				selector string
+				pod      string
+			}{
+				{
+					selector: "id=app1",
+					pod:      fe1Name,
+				},
+				{
+					selector: "id=app2",
+					pod:      fe2Name,
+				},
+			}
+			for _, tc := range testCases {
+				seenBe1 := false
+				seenBe2 := false
+				for try := 0; try < repeat; try++ {
+					res, err := kubectl.ExecInPods(ctx, helpers.DefaultNamespace, tc.selector, curlTCP)
+					Expect(err).To(BeNil(), "http curl failed in %s pod", tc.selector)
+					if strings.Contains(res[tc.pod].Stdout(), be1Name) {
+						seenBe1 = true
+					}
+					if strings.Contains(res[tc.pod].Stdout(), be2Name) {
+						seenBe2 = true
+					}
+					if seenBe1 && seenBe2 {
+						break
+					}
+				}
+				// Expects to see both backends (local and remote) to appear
+				Expect(seenBe1 && seenBe2).To(BeTrue(), "backend reachable:\n%s\t%v\n%s\t%v", be1Name, seenBe1, be2Name, seenBe2)
```

Comment:
Maybe we don't need to do a boolean AND here. Probably an individual assertion on both should be sufficient, and removes the need to pretty print the error msg.

Reply:
Good point, done :)